storage/storage_transport.go

```diff
@@ -0,0 +1,77 @@
+package storage
+
+import (
+	"errors"
+	"regexp"
+
+	"github.com/Sirupsen/logrus"
+	"github.com/containers/image/types"
+	"github.com/containers/storage/storage"
+)
+
+var (
+	// Transport is an ImageTransport that uses a default storage.Mall or
+	// one that's it's explicitly told to use.
+	Transport MallTransport = &storageTransport{}
+	// ErrUnknownPolicyConfigurationScope is returned when
+	// ValidatePolicyConfigurationScope() fails.
+	ErrUnknownPolicyConfigurationScope = errors.New("unknown policy configuration scope")
+	// ErrInvalidReference is returned when ParseReference() is passed an
+	// empty reference.
+	ErrInvalidReference = errors.New("invalid reference")
+	idRegexp            = regexp.MustCompile("^(sha256:)?([0-9a-fA-F]{64})$")
+)
+
+// MallTransport is an ImageTransport that uses a storage.Mall, either its own
+// or one that it's told to use.
+type MallTransport interface {
+	types.ImageTransport
+	SetMall(storage.Mall) error
+}
+
+type storageTransport struct {
+	mall storage.Mall
+}
+
+func (s *storageTransport) Name() string {
+	return "oci-storage"
+}
+
+// SetMall sets the Mall object which the Transport will use.  If one is not
+// explicitly provided, the library will attempt to initialize one with default
+// settings the first time a reference needs to be parsed.
+func (s *storageTransport) SetMall(mall storage.Mall) error {
```

Comment:
Would this change the semantics of any existing `ImageReference` objects based of this `storageTransport`?

My first impression is that perhaps the `Mall` should be an attribute of the `ImageReference`, or the `storageTransport` should be initialized with a `Mall` and be immutable from that point on.  But then I have no idea what a `Mall` is :)